add Turbine DTO

adamkorynta opened the pull request:
added unit tests for serialization and validation

adamkorynta committed:
add Turbine DTO

added unit tests for serialization and validation

adamkorynta commented:
Once PR: https://github.com/USACE/cwms-data-api/pull/705 is merged, I'll take advantage of the LocationIdentifier to couple together the project-id and project-office-id

adamkorynta commented:
Rebased on Bryson's LocationIdentifier change and incorporated

adamkorynta commented:
Adding accept header aliasing in controller/dao PR: https://github.com/USACE/cwms-data-api/pull/717

rma-psmorris commented on cwms-data-api/src/test/resources/cwms/cda/data/dto/location/kind/turbine.json:
```diff
@@ -0,0 +1,28 @@
+  },
+  "location": {
+    "office-id": "LRL",
+    "name": "TEST_LOCATION2",
```
Why is this "name" here when in the project identifier it is "location-id"? Can we use one term for both?

adamkorynta replied:
good point, I'll update LocationIdentifier to use name in order to match other references.

adamkorynta replied:
updated

adamkorynta committed:
Update `location-id` field to `name` for `LocationIdentifier`

provides consistency with the `Location` DTO. Also renamed `project-identifier` to match the `Stream*` DTO's for consistency.

adamkorynta committed:
remove hashcode/equals from LocationIdentifier/Turbine/Embankment

logic was only used in test code, moved logic to only test code

adamkorynta committed:
add back in identifier equality

rma-psmorris reviewed (approved):
Reviewed with focus on DTO/JSON.

rma-rripken commented on cwms-data-api/src/test/resources/cwms/cda/data/dto/location/kind/embankment.json:
```diff
@@ -49,5 +25,29 @@
+  "height-max": 5.0,
+  "top-width": 20.0,
+  "units-id": "ft",
+  "downstream-prot-type": {
```
We're good with "prot" as an abbreviation "term of art" etc.   I'm fine with it.  Making sure there is concensus.

MikeNeilson replied:
what the heck is "prot"?

adamkorynta replied:
stands for "protection", I was following db and java code. I'll revise in a separate PR (trying to keep this to turbines only)

adamkorynta replied:
and failing....

MikeNeilson replied:
Okay, we should definitely spell that out, but it can wait for now.

rma-rripken commented on cwms-data-api/src/test/resources/cwms/cda/data/dto/location/kind/embankment.json:
```diff
@@ -49,5 +25,29 @@
+  "upstream-side-slope": 15.0,
+  "downstream-side-slope": 90.0,
+  "structure-length": 25.0,
+  "height-max": 5.0,
```
Why is it height-max and not max-height?

adamkorynta replied:
huh, looks like every other usage of height+max in the database is `max_height`. I'll update
